c/driver/bigquery/adbc-driver-bigquery.pc.in

```diff
@@ -23,3 +23,4 @@ Description: The ADBC BigQuery driver provides an ADBC driver for BigQuery.
 URL: https://github.com/apache/arrow-adbc
 Version: @ADBC_VERSION@
 Libs: -L${libdir} -ladbc_driver_bigquery
+Libs.private: -DADBC_STATIC
```

Comment:
I don't think the modules built by Go (bigquery, flightsql, snowflake) would ever be static (?) so perhaps this is superfluous, but I figured worth keeping for consistency with other .pc files

Reply:
See https://github.com/apache/arrow-adbc/pull/2738, they can be built and used as static libraries, but there's caveats as the static library contains the Go runtime

Reply:
Ah OK great - with how this PR has evolved its probably worth merging that one first and rebasing this on top